### PR TITLE
core: rename diagnostics to debugData

### PR DESCRIPTION
**Pull request by patrickhulce**
**Summary**
All the data here is essentially used for debugging, so `debugData` seems fair.

`metrics` audit is probably the one exception where I'd expect folks to be explicitly relying on this for presentation, so I'm not sure what to do there. OTOH I also don't feel particularly strongly it needs to be renamed away from `diagnostics` since neither of them describe how it's being used in `metrics` :)

**Related Issues/PRs**
No issue was filed specifically, so it can't be closed yet! ;)
#7752

### CODE REVIEW

File: lighthouse-core/audits/accessibility/axe-audit.js [hunk 1]
```diff
@@ -63,11 +63,11 @@ class AxeAudit extends Audit {
       {key: 'node', itemType: 'node', text: str_(UIStrings.failingElementsHeader)},
     ];
 
-    /** @type {LH.Audit.Details.Diagnostic|undefined} */
+    /** @type {LH.Audit.Details.DebugData|undefined} */
     let diagnostic;
```
Comment (brendankenny) [inline lighthouse-core/audits/accessibility/axe-audit.js]: rename variable?

File: types/audit-details.d.ts [hunk 2]
```diff
@@ -51,7 +51,7 @@ declare global {
         overallSavingsBytes?: number;
         headings: OpportunityColumnHeading[];
         items: OpportunityItem[];
-        diagnostic?: Diagnostic;
+        diagnostic?: DebugData;
```
Comment (brendankenny) [inline types/audit-details.d.ts]: I think all these `diagnostic` need to be renamed?

File: lighthouse-core/audits/byte-efficiency/uses-long-cache-ttl.js [hunk 3]
```diff
@@ -238,18 +238,18 @@ class CacheHeaders extends Audit {
         if (url.includes('?')) queryStringCount++;
 
         // Include cacheControl info (if it exists) per url as a diagnostic.
-        /** @type {LH.Audit.Details.Diagnostic|undefined} */
-        let diagnostic;
+        /** @type {LH.Audit.Details.DebugData|undefined} */
+        let debugData;
```
Comment (brendankenny) [inline lighthouse-core/audits/byte-efficiency/uses-long-cache-ttl.js]: just `debug`?

File: types/audit-details.d.ts [hunk 4]
```diff
@@ -68,15 +68,15 @@ declare global {
...
 
       /**
        * A details type that is not rendered in the final report; usually used
-       * for including diagnostic information in the LHR. Can contain anything.
+       * for including debug information in the LHR. Can contain anything.
        */
-      export interface Diagnostic {
-        type: 'diagnostic';
+      export interface DebugData {
+        type: 'debug';
```
Comment (brendankenny) [inline types/audit-details.d.ts]: if we go with type `DebugData` maybe this string should be `debugdata` and the property name `debugData`?

It's a little confusing having both `debug` and `debugData` as reasonable local variable names

Comment (patrickhulce) [reply types/audit-details.d.ts]: agreed, `debugdata` SGTM :)

File: types/audit-details.d.ts (hunk 2 above)
Comment (patrickhulce) [reply types/audit-details.d.ts]: yeah fo sho, didn't expect the first name go around to be accepted :D

**Comment by paulirish** [comment]
name wfm. :+1:

**Pull request merged**